tjpcov_new/covariance_io.py

```diff
@@ -0,0 +1,151 @@
+import pyccl as ccl
+import pickle
+import sacc
+import numpy as np
+import warnings
+import yaml
+import jinja2
+import importlib
+import os
+
+class CovarianceIO():
+    def __init__(self, config):
+        """
+        CovarianceIO class for TJPCov.
+
+        This is a base class that handles the input/output of the covariances.
+        It does not computation.
+
+        Parameters
+        ----------
+        config (dict or str):
+        """
+        self.config = self._read_config(config)
+        self.sacc_file = None
+
+        # Output directory where to save all the time consuming calculations
+        self.outdir = self.config['tjpcov'].get('outdir', './')
+
+    def _read_config(self, config):
+        if isinstance(config, dict):
+            pass
+        elif isinstance(config, str):
+            config = self._parse(config)
+        else:
+            raise ValueError("config must be of type dict or str, given" +
+                             f"{type(config)}")
+
+        return config
+
+    @staticmethod
+    def _parse(filename):
+        """
+        Parse a configuration file.
+
+        Parameters
+        ----------
+        filename : str
+            The config file to parse. Should be YAML formatted.
+
+        Returns
+        -------
+        config: dict
+            The raw config file as a dictionary.
+        """
+
+        with open(filename, 'r') as fp:
+            config_str = jinja2.Template(fp.read()).render()
+        config = yaml.load(config_str, Loader=yaml.Loader)
+
+        return config
+
+    def create_sacc_cov(self, output='cls_cov.fits'):
+        """
+        Write created cov to a new sacc object
+
+        Parameters:
+        ----------
+        output (str): filename output.
+
+        Returns:
+        -------
+        None
+
+        """
+        output = os.path.join(self.get_outdir(), output)
+
+        cov = self.get_covariance()
+
+        s = self.get_sacc_file().copy()
+        s.add_covariance(cov)
+        s.save_fits(output, overwrite=True)
+
+        return s
+
+    def get_covariance(self):
+        raise NotImplementedError("Do not use the base class directly")
```

Comment:
This all looks great @carlosggarcia !  One thing you can do here is actually make `CovarianceIO` inherit from `ABC` (Abstract Base Class) e.g. `CovarianceIO(ABC)`.  Then you can use the `@abstractmethod` decorator on `get_covariance(self)`.  It would look like

```
@abstractmethod
def get_covariance(self):
    pass
```

So you're basically declaring the method signature for the abstract method with no body.  Could be a little cleaner than raising errors, but either works, just a small suggestion!

Reply:
This looks good! Thanks!